lib/gears/timer.lua

```diff
@@ -88,7 +88,21 @@ function timer:start()
         print(traceback("timer already started"))
         return
     end
-    self.data.source_id = glib.timeout_add(glib.PRIORITY_DEFAULT, self.data.timeout * 1000, function()
+    local func, timeout
+    if self.data.precise == nil then
+        self.data.precise = select(2, math.modf(self.data.timeout)) ~= 0
```

Comment:
Sorry, but no.

First, this is nuts once you write the docs for it:

> The default value for `precise` depends on the timeout that the timer is started with for the first time. If that timeout has no fractional value (i.e. is an integer), the default is `true`. Otherwise, the default is `false`.

Also, isn't this a behaviour change that we are not supposed to do in stable releases?